db/fetcher/fetcher.go

```diff
@@ -606,16 +621,46 @@ func (df *DocumentFetcher) fetchNext(ctx context.Context) (EncodedDocument, Exec
 			}
 		}
 
+		// Check if we have read access, for document on this collection, with the given identity.
+		if !df.passedPermissionCheck {
```

Comment:
todo: you are running the `CheckAccess` on every KV, but it only applies to the Document as a whole. Need to run once, likely at the beginning doc loop, when we get the first KV of a doc (which automatically populates the DocID, which is the only info you need to run this)

Reply:
I am not sure if that is true, this if condition will be entered once (initially `passedPermissionCheck` is false). Then after the `check` either:

1) ```passedPermissionCheck == true```: Then can never enter this again, hence no more CheckAccess calls.

2) ```passedPermissionCheck == false```: Then there is no point in collecting other fields so it will seek next doc.

In both those cases only one check is made.

Reply:
I can document that better however. In the code.

Reply:
- [ ] Document that if no-access to document then will seek next document instead of collecting other fields.

Reply:
> In both those cases only one check is made.

Yupp you are indeed correct, took me a second to wire it all back in my brain. Apologies, I've spent too long in the `fetcher` package, and I'm a bit sensitive I guess 🤡

Reply:
Cheers, in that case resolving please re-raise incase you want to continue or have another todo item.